Mathlib/Analysis/Analytic/IsolatedZeros.lean

```diff
@@ -217,6 +217,27 @@ lemma order_eq_nat_iff (hf : AnalyticAt 𝕜 f z₀) (n : ℕ) : hf.order = ↑n
     refine ⟨fun hn ↦ (WithTop.coe_inj.mp hn : h.choose = n) ▸ h.choose_spec, fun h' ↦ ?_⟩
     rw [unique_eventuallyEq_pow_smul_nonzero h.choose_spec h']
 
+/- An analytic function `f` has finite order at a point `z₀` iff locally looks
+  like `(z - z₀) ^ order • g`, where `g` is analytic and does not vanish at
+  `z₀`. -/
+lemma order_neq_top_iff (hf : AnalyticAt 𝕜 f z₀) :
+    hf.order ≠ ⊤ ↔ ∃ (g : 𝕜 → E), AnalyticAt 𝕜 g z₀ ∧ g z₀ ≠ 0
+      ∧ f =ᶠ[𝓝 z₀] fun z ↦ (z - z₀) ^ (hf.order.toNat) • g z := by
+  erw [← hf.order_eq_nat_iff]
```

Comment:
In general, `erw` is a code smell: usually it indicates something sub-optimal.
Can you tell what `rw`s you need to use? (Maybe there are even some missing API lemmas.)

If you get stuck, feel free to ask for help here or on zulip.

Reply:
The issue is that the lemma `order_neq_top_iff` speaks about `f =ᶠ[𝓝 z₀]  F` while the theorem `order_eq_nat_iff` speaks about `∀ᶠ (z : 𝕜) in 𝓝 z₀, f z = F z`. These two statements are definitionally equal (that´s the definition of `EventuallyEq`) but the `rw` and `simp_rw` tactics are not able to make the match. I have replaced the offending line by `unfold EventuallyEq; rw [← hf.order_eq_nat_iff]`, but one could think about adding a theorem `EventuallyEq.def` that can be used with `rw`. Please let me know if you want me to add that.

Reply:
Does `rw [EventuallyEq]` work here?

Reply:
Implemented, following David's suggestion below.